.github/workflows/windows.yml

```
@@ -413,52 +413,125 @@ jobs:
           cmake --build bld --config '${{ matrix.type }}' --target test-ci
 
   msvc:
-    name: 'msvc (${{ matrix.arch }}, ${{ matrix.plat }}, ${{ matrix.config }})'
+    name: 'msvc (${{ matrix.name }})'
     runs-on: windows-latest
     timeout-minutes: 30
     strategy:
       matrix:
         include:
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=ON  -DBUILD_SHARED_LIBS=OFF -DENABLE_UNICODE=ON ', type: 'Debug' }
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=OFF -DBUILD_SHARED_LIBS=OFF -DENABLE_UNICODE=OFF', type: 'Debug' }
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=OFF -DHTTP_ONLY=ON          -DENABLE_UNICODE=OFF', type: 'Debug' }
+        # name should be unique and without special chars. It create a folder with that name for cache the libreries.
+        - name: 'schannel'
+          install: 'nghttp2 zstd zlib'
+          arch: 'x64'
+          plat: 'windows'
+          type: 'Debug'
+          tflags: '~1516'
+          config: '-DBUILD_SHARED_LIBS=OFF -DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DCURL_USE_SCHANNEL=ON -DENABLE_UNICODE=ON'
+        - name: 'openssl'
+          install: 'nghttp2 nghttp3 zstd libssh2'
+          arch: 'x64'
+          plat: 'windows'
+          type: 'Debug'
+          tflags: '~1516'
+          config: '-DBUILD_SHARED_LIBS=OFF -DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DENABLE_UNICODE=ON -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DUSE_OPENSSL_QUIC=ON'
+        - name: 'no ssl'
+          install: 'nghttp2 zstd zlib'
+          arch: 'x64'
+          plat: 'windows'
+          type: 'Debug'
+          tflags: '~1516'
+          config: '-DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DCURL_USE_SCHANNEL=OFF -DHTTP_ONLY=ON  -DENABLE_UNICODE=OFF'
       fail-fast: false
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: 'retore cache dependencies'
+        uses: actions/cache/restore@v4
+        with:
+          path: 'cache/${{ matrix.name }}'
+          key: ${{ matrix.plat }}-${{ matrix.name }}
+          restore-keys: ${{ matrix.plat }}-${{ matrix.name }}
+
+      - name: 'vcpkg init'
+        if: success()
+        shell: bash
+        run: |
+          git clone https://github.com/microsoft/vcpkg/ --depth 1
+          vcpkg/bootstrap-vcpkg.sh
+
+      - name: 'vcpkg build'
```

```suggestion
      - name: 'vcpkg build'
      - timeout-minutes: 5
```
Or any value you find fitting here. (Seems to take 30 seconds in current runs?)

---

5 min it wrong. 
It can be 10min.
On the worst case that library is updated, it need to compile it from source, and it take time.
It can be more then 10 when we add different libs. I will change this to 10 min.

---

10 mins sounds fine; we can adjust it as needed. Point is to avoid a runaway (or hanging) step.